// python -m grpc_tools.protoc --python_out=. --pyi_out=. --grpc_python_out=. -I./ file_service.proto
syntax = "proto3";

// The greeting service definition.
service Greeter {
  // Sends a greeting
  // Uploads a file
  rpc UploadFile(File) returns (MetaData) {}
  // Downloads a file
  rpc DownloadFile(MetaData) returns  (File) {}

  rpc RemoveFile(MetaData) returns (MetaData) {}

  rpc GetFileList(FileListRequest) returns (FileListResponse) {}

  rpc GetFile(FileRequest) returns (MetaData) {}
}


message File {
  optional MetaData meta = 1;
  bytes chunk_data = 2;
}

message FileRequest {
  string bucket = 1;
  string filename = 2;
  string extension = 3;
}

message MetaData {
  optional string bucket = 1;
  string filename = 2;
  string extension = 3;
  optional string hash = 4;
  optional string date = 5;
}

message FileListRequest {
  string bucket = 1;
}

message FileListResponse {
  repeated MetaData files = 1;
}
